### DISCUSSION TITLE
Accept only one line, in vim

### DISCUSSION THREAD

**Discussion by i40west** [community/community#50196] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Sometimes a Copilot suggestion starts out great, then goes off the rails, line after line, but accepting it is all-or-nothing. It would be great to be able to accept only one line.

So I [forked the repo](https://github.com/i40west/copilot.vim) and added this feature. But, for ease of merging future updates, I did it by implementing a function almost identical to the `Accept` function, which begs for refactoring, and binding it to shift-tab. Also, I've never written a line of vimscript before except my .vimrc file. So you won't want my version, but it would be nice to have this as an actual feature.

**Comment by web-sst** [comment]
Can this be extended to allow selection by text object that matches in the suggestion? I frequently want to accept n words, or the first of two sentences in addition to a single line.

**Reply by MunifTanjim** [reply]
You can implement that using `copilot.lua`: https://github.com/zbirenbaum/copilot.lua/issues/151#issuecomment-1514782725

**Comment by shoxter** [comment]
I can't even get mine to ever suggest more than one line at a time. Very frustrating (and even unusable) when in a LISP language.
